packages/salesforcedx-vscode-core/src/commands/forcePackageInstall.ts

```diff
@@ -0,0 +1,156 @@
+import {
+  CancelResponse,
+  ContinueResponse,
+  ParametersGatherer
+} from '@salesforce/salesforcedx-utils-vscode/out/src/types';
+import { Observable } from 'rxjs/Observable';
+import * as vscode from 'vscode';
+import { channelService } from '../channels';
+import { PKG_ID_PREFIX } from '../constants';
+import { nls } from '../messages';
+import { notificationService, ProgressNotification } from '../notifications';
+import { taskViewService } from '../statuses';
+import { getRootWorkspacePath } from '../util';
+import {
+  CompositeParametersGatherer,
+  EmptyPreChecker,
+  SfdxCommandlet,
+  SfdxCommandletExecutor
+} from './util';
+
+type forcePackageInstallOptions = {
+  packageId: string;
+  installationKey: string;
+};
+
+export class ForcePackageInstallExecutor extends SfdxCommandletExecutor<
+  PackageIdAndInstallationKey
+> {
+  private readonly options: forcePackageInstallOptions;
+
+  public constructor(options = { packageId: '', installationKey: '' }) {
+    super();
+    this.options = options;
+  }
+
+  public build(data: PackageIdAndInstallationKey): Command {
+    const builder = new SfdxCommandBuilder()
+      .withDescription(nls.localize('force_package_install_text'))
+      .withArg('force:package:install')
+      .withFlag('--package', data.packageId)
+      .withLogName('force_package_install');
+
+    if (data.installationKey) {
+      builder.withFlag('--installationkey', data.installationKey);
+    }
+
+    return builder.build();
+  }
+
+  public execute(
+    response: ContinueResponse<PackageIdAndInstallationKey>
+  ): void {
+    const startTime = process.hrtime();
+    const cancellationTokenSource = new vscode.CancellationTokenSource();
+    const cancellationToken = cancellationTokenSource.token;
+
+    const execution = new CliCommandExecutor(this.build(response.data), {
+      cwd: getRootWorkspacePath(),
+      env: { SFDX_JSON_TO_STDOUT: 'true' }
+    }).execute(cancellationToken);
+
+    notificationService.reportExecutionError(
+      execution.command.toString(),
+      (execution.stderrSubject as any) as Observable<Error | undefined>
+    );
+    channelService.streamCommandOutput(execution);
+    ProgressNotification.show(execution, cancellationTokenSource);
+    taskViewService.addCommandExecution(execution, cancellationTokenSource);
+  }
+}
+
+export type PackageIdAndInstallationKey = PackageID & InstallationKey;
+
+export interface PackageID {
+  packageId: string;
+}
+
+export interface InstallationKey {
+  installationKey: string;
+}
+
+export class SelectPackageID implements ParametersGatherer<PackageID> {
+  public async gather(): Promise<CancelResponse | ContinueResponse<PackageID>> {
+    const packageIdInputOptions = {
+      prompt: nls.localize('parameter_gatherer_enter_package_id'),
+      placeHolder: nls.localize('package_id_gatherer_placeholder'),
+      validateInput: value => {
+        return isRecordId(value, PKG_ID_PREFIX) || value === ''
```

Comment:
Will `value === ''` permit entering a blank package id?

Reply:
No, the command validation with catch that and not let the user continue